src/control/server/ctl_ranks_rpc_test.go

```diff
@@ -293,27 +279,28 @@ func TestServer_CtlSvc_StopRanks(t *testing.T) {
 			signalErr: errors.New("sending signal failed"),
 			expErr:    errors.New("sending killed: sending signal failed"),
 		},
-		"context timeout": { // near-immediate parent context Timeout
-			req:        &ctlpb.RanksReq{Ranks: "0-3"},
-			ctxTimeout: time.Millisecond,
-			expErr:     context.DeadlineExceeded, // parent ctx timeout
-		},
-		"instances started": { // unsuccessful result for kill
+		"instances successfully stopped": {
 			req:            &ctlpb.RanksReq{Ranks: "0-3"},
 			expSignalsSent: map[uint32]os.Signal{0: syscall.SIGINT, 1: syscall.SIGINT},
 			expResults: []*sharedpb.RankResult{
-				{Rank: 1, State: msErrored, Errored: true},
-				{Rank: 2, State: msErrored, Errored: true},
+				{Rank: 1, State: msStopped},
+				{Rank: 2, State: msStopped},
 			},
 		},
-		"force stop instances started": { // unsuccessful result for kill
+		"instances successfully stopped with force": {
 			req:            &ctlpb.RanksReq{Ranks: "0-3", Force: true},
 			expSignalsSent: map[uint32]os.Signal{0: syscall.SIGKILL, 1: syscall.SIGKILL},
 			expResults: []*sharedpb.RankResult{
-				{Rank: 1, State: msErrored, Errored: true},
-				{Rank: 2, State: msErrored, Errored: true},
+				{Rank: 1, State: msStopped},
+				{Rank: 2, State: msStopped},
 			},
 		},
+		"instances not stopped in time": {
+			req:               &ctlpb.RanksReq{Ranks: "0-3"},
+			expSignalsSent:    map[uint32]os.Signal{0: syscall.SIGINT, 1: syscall.SIGINT},
+			instancesDontStop: true,
+			expErr:            errors.New("deadline exceeded"),
+		},
```

Comment:
I don't understand why we have any test cases involving timeouts now. There is no production code setting a server-side timeout, so I'm not seeing why there should be a test to exercise this.

If the intent is to verify that the code correctly handles a canceled context, then I suggest that you just create a context with a 1ns deadline so that you're guaranteed to hit it.

Reply:
Example of a case where I think it's needed:
- attempting to test situation where rank doesn't start when it's asked to
- we configure the engine to ignore the request to start
- we need to evaluate that the engine has not started after a given amount of time
- if we don't apply a time out in the test context then the test runs for the entire outer test timeout (~30s) before failing

Does that make sense?

Reply:
I think in the case of a rank not starting, we should be returning a better error message, something like: "rank(s) [1, ..., n] didn't start"

Reply:
with the change of removing the inner timeouts this is now not possible, you don't have time to retrieve context of failure after the outer RPC context has been cancelled